olive/passes/onnx/transformer_optimization.py

```diff
@@ -217,10 +217,12 @@ def _run_for_config(
                 "OrtTransformersOptimization.config"
             )
 
-        output_model_path = ONNXModel.resolve_path(os.path.join(output_model_path, os.path.basename(model.model_path)))
+        output_model_path = Path(output_model_path)
+        if output_model_path.suffix != ".onnx":
+            output_model_path = output_model_path / Path(model.model_path).name
+        output_model_path = ONNXModel.resolve_path(output_model_path)
```

Comment:
Actually, I think line 220-223 is equivalent to 
```
output_model_path = ONNXModel.resolve_path(output_model_path, Path(model.model_path).name)
```
`resolve_path` already handles the suffix check https://github.com/microsoft/Olive/blob/main/olive/model/__init__.py#L312

Reply:
That would be two consecutive calls to `resolve_path` which is confusing, in my opinion. This is more explicit.

Reply:
it's just one. all of lines 220-223 is replaces by this single line

Reply:
Ahh, I get that. I will switch to that in the PR #747